packages/sources/layer2-sequencer-health/src/starkware.ts

```diff
@@ -0,0 +1,118 @@
+import { race } from './network'
+
+interface PendingBlockState {
+  parentBlockHash: string
+  txnCount: number
+  lastUpdated: number
+  lastStatus: boolean
+}
+
+export const sendDummyStarkwareTransaction = async (config: ExtendedConfig): Promise<boolean> => {
+  const starkKeyPair = ec.genKeyPair(DEFAULT_PRIVATE_KEY)
+  const starkKeyPub = ec.getStarkKey(starkKeyPair)
+  const provider = config.starkwareConfig.provider
+  const account = new Account(provider, config.starkwareConfig.argentAccountAddr, starkKeyPair)
+
+  const receipt = await race<AddTransactionResponse>({
+    timeout: config.timeoutLimit,
+    promise: account.execute(
+      {
+        contractAddress: config.starkwareConfig.argentAccountAddr,
+        entrypoint: 'initialize',
+        calldata: [starkKeyPub, '0'],
+      },
+      undefined,
+      { maxFee: '0' },
+    ),
+    error: `Transaction receipt not received in ${config.timeoutLimit} milliseconds`,
+  })
+  Logger.info(
+    `Transaction receipt received with hash ${receipt.transaction_hash} for EVM network: ${Networks.Starkware}`,
+  )
+  await provider.waitForTransaction(receipt.transaction_hash)
+  const tx = await provider.getTransaction(receipt.transaction_hash)
+  return tx.status !== 'REJECTED'
+}
+
+/**
+ * The centralized Starkware sequencer is made up of two internal components.  The first is the gateway which is used to query the Sequencer state
+ * and the batcher which adds transactions to the pending block and processes the pending block to add it into the L2 chain.  The
+ * Sequencer is considered unhealthy if either of the components are down.
+ *
+ *  - The gateway is considered to be down if request to fetch pending block times out or if we get a 404
+ *  - The batcher is considered to be down if we a new block has not been produced AND no new transactions have been added to the
+ *  pending block within a given time interval.
+ * status when fetching
+ * @param networks
+ * @returns true if Sequencer is healthy
+ */
+export const checkStarkwareSequencerPendingTransactions = (): ((
+  config: ExtendedConfig,
+) => Promise<boolean>) => {
+  let lastSeenPendingBlock: PendingBlockState = {
+    parentBlockHash: '',
+    txnCount: 0,
+    lastUpdated: 0,
+    lastStatus: true,
+  }
+  return async (config: ExtendedConfig): Promise<boolean> => {
+    const delta = config.delta
+    const currentTime = Date.now()
+    if (
+      lastSeenPendingBlock.lastUpdated > 0 &&
+      currentTime - lastSeenPendingBlock.lastUpdated < delta
+    ) {
+      Logger.debug(
+        `Skipping check for Starkware Sequencer health as it has been less than ${delta} seconds since last check`,
+      )
+      return lastSeenPendingBlock.lastStatus
+    }
+    let pendingBlockParams: GetBlockResponse
+    try {
+      pendingBlockParams = await config.starkwareConfig.provider.getBlock('pending')
+    } catch (e: any) {
+      if (e.providerStatusCode === 504) {
```

Comment:
We should retry 2 times with some backoff, before marking it unhealthy. It could just be flaky.